Fixed anchor sanitization and the failing fonts test

arzola opened the pull request:
Related issue #2088 

### Problem

When links are being saved all the attributes quotes are being scaped this was not allowing`HtmLawed` to recognize a valid href schema.

### Solution

I added a `wp_kses_stripslashes` on the sanitization function that will only remove scaped quotes in html attributes that needs to be sanitized

Also two unit test assertions were added

### How to test

1. Try to save an anchor link into the fields allows HTML with some XSS strings
2. Wellformed urls will be saved in the href attribute other malicious tags should be striped

richard015ar commented:
@arzola I tested this, and I had some problems by adding some links surrounded by `'` instead `"`.
For example: `<a href='https://google.com'>I am a link</a>`. It doesn't work as expected since it removes the `href` value completely.
Also, we need define if we will allow links in the short description as Amy mentioned in the ticket. If that the case, I guess you need to implement it in this PR.

There are a few names of variables that could be improved in the new tests you added, but it is not mandatory, just to keep your policy about naming conventions.

richard015ar reviewed (changes requested):
@arzola I tested this, and I had some problems by adding some links surrounded by `'` instead `"`.
For example: `<a href='https://google.com'>I am a link</a>`. It doesn't work as expected since it removes the `href` value completely.
Also, we need define if we will allow links in the short description as Amy mentioned in the ticket. If that the case, I guess you need to implement it in this PR.

There are a few names of variables that could be improved in the new tests you added, but it is not mandatory, just to keep your recommendations about naming conventions.

arzola commented:
@richard015ar hey bro, I fixed the issue with single quotes, just one thing the library is going to convert single quotes to double quotes.

I added support for HTML in the short description field.

For the naming conventions I was trying to follow the current "wordpress" standards snake_case according to https://github.com/pressbooks/coding-standards is there something that I'm doing wrong?

Thank you for your feedback

richard015ar commented:
LGTM! easy and elegant solution!
Thank you Oscar!
